File: src/utils/ansi.ts
/**
 * 极简 ANSI 颜色 + 光标控制（不引 chalk，零依赖）。
 * 非 TTY 或设了 NO_COLOR 时自动退化为无色，方便管道/重定向。
 */
const enabled = Boolean(process.stdout.isTTY) && !process.env.NO_COLOR;

const wrap = (open: number, close: number) => (s: string): string =>
  enabled ? `\x1b[${open}m${s}\x1b[${close}m` : s;

export const green = wrap(32, 39);
export const yellow = wrap(33, 39);
export const cyan = wrap(36, 39);
export const red = wrap(31, 39);
export const dim = wrap(2, 22);
export const bold = wrap(1, 22);

export type Color = 'green' | 'yellow' | 'cyan' | 'red' | 'dim' | 'bold' | 'none';

export function paint(s: string, color: Color): string {
  switch (color) {
    case 'green':
      return green(s);
    case 'yellow':
      return yellow(s);
    case 'cyan':
      return cyan(s);
    case 'red':
      return red(s);
    case 'dim':
      return dim(s);
    case 'bold':
      return bold(s);
    default:
      return s;
  }
}

// —— 光标 / 屏幕控制 ——
export const HIDE_CURSOR = '\x1b[?25l';
export const SHOW_CURSOR = '\x1b[?25h';

/** 光标上移 n 行（n<=0 时为空串）。 */
export const cursorUp = (n: number): string => (n > 0 ? `\x1b[${n}A` : '');

/** 从光标处清到屏幕末尾。 */
export const CLEAR_DOWN = '\x1b[0J';

/** 回到行首。 */
export const CR = '\r';

/** 清屏 + 清回滚 + 光标归位（等价于 PowerShell 的 Clear-Host，制造「整页」感）。 */
export const CLEAR_SCREEN = '\x1b[2J\x1b[3J\x1b[H';

File: internal/tui/ansi.go
// Package tui 是自绘 ANSI 终端 UI：raw-key 菜单 + cooked 文本输入 + 终端原语抽象。
// 移植自 npm 版 src/ui/*，是 Go 版最大风险区（raw mode / 中文输入法 / TTY 恢复 / Ctrl+C / spawn 继承）。
package tui

import (
	"os"
	"strconv"

	"golang.org/x/term"
)

// colorEnabled：仅当 stdout 是终端且未设 NO_COLOR 时上色（管道/重定向自动退化为无色）。
var colorEnabled = term.IsTerminal(int(os.Stdout.Fd())) && os.Getenv("NO_COLOR") == ""

// Color 是支持的颜色名。
type Color string

const (
	ColorNone   Color = "none"
	ColorGreen  Color = "green"
	ColorYellow Color = "yellow"
	ColorCyan   Color = "cyan"
	ColorRed    Color = "red"
	ColorDim    Color = "dim"
	ColorBold   Color = "bold"
)

var colorCodes = map[Color][2]int{
	ColorGreen:  {32, 39},
	ColorYellow: {33, 39},
	ColorCyan:   {36, 39},
	ColorRed:    {31, 39},
	ColorDim:    {2, 22},
	ColorBold:   {1, 22},
}

// Paint 给字符串上色（无色或未知色原样返回）。
func Paint(s string, c Color) string {
	code, ok := colorCodes[c]
	if !ok || !colorEnabled {
		return s
	}
	return "\x1b[" + strconv.Itoa(code[0]) + "m" + s + "\x1b[" + strconv.Itoa(code[1]) + "m"
}

// —— 光标 / 屏幕控制 ——
const (
	HideCursor  = "\x1b[?25l"
	ShowCursor  = "\x1b[?25h"
	ClearDown   = "\x1b[0J"              // 从光标处清到屏幕末尾
	CR          = "\r"                   // 回到行首
	ClearScreen = "\x1b[2J\x1b[3J\x1b[H" // 清屏 + 清回滚 + 归位（制造整页感）
)

// CursorUp 光标上移 n 行（n<=0 返回空串）。
func CursorUp(n int) string {
	if n > 0 {
		return "\x1b[" + strconv.Itoa(n) + "A"
	}
	return ""
}
